### PR TITLE
Integrate link graph into main wiki pipeline

### PR DESCRIPTION
**Pull request by ringgaard**
I have integrated the link graph into the main wiki pipeline. The fanin values now also includes counts from basic facts from the items.

I have also updated the documentation to use the new Myelin-based parser trainer.

### CODE REVIEW

File: sling/nlp/wiki/wikipedia-links.cc [hunk 1]
```diff
@@ -157,6 +158,46 @@ class WikipediaLinkExtractor : public task::DocumentProcessor {
...
+      if (slot.name == Handle::isa()) continue;
+      if (slot.name == n_lang_) continue;
+
+      Handle target = store->Resolve(slot.value);
+      if (!store->IsFrame(target)) continue;
+
+      Text id = store->FrameId(target);
+      if (id.empty()) continue;
+
+      if (!store->IsFrame(target)) continue;
```
Comment (anders-sandholm) [inline sling/nlp/wiki/wikipedia-links.cc]: Looks like this line might be redundant.

File: sling/nlp/wiki/wikipedia-links.cc [hunk 2]
```diff
@@ -157,6 +158,46 @@ class WikipediaLinkExtractor : public task::DocumentProcessor {
...
+       accumulator_.Increment(id);
```
Comment (anders-sandholm) [inline sling/nlp/wiki/wikipedia-links.cc]: indention off by one

File: doc/guide/caspar.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+[PropBank](https://propbank.github.io/) semantic role labels are extracted from 
+this corpus to produce a frame semantic corpus in SLING document format.
+
+We use the standard [CoNLL-2012](http://conll.cemantix.org/2012/data.html) split
+of the data to produce training, development, and test corpora.
+
+## Preparing the training data
+
+The LDC2013T19 OntoNotes 5 corpus is needed to produce the training data for
+CASPAR. This is licensed by LDC and you need a LDC license to use the corpus:
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply doc/guide/caspar.md]: Done

File: doc/guide/caspar.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+
+This will put the training and evaluation data into `local/data/corpora/caspar`:
+* `train.rec` contains the training data.
+* `dev.rec` contains the developemnt data.
+* `test.rec` contains the evaluation data.
+* `train_shuffled.rec` contains the shuffled training data.
+
+## Pre-trained word embeddings
+
+The CASPAR parser uses pre-trained word embeddings which can be download from
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply doc/guide/caspar.md]: Done.

File: doc/guide/install.md [hunk 5]
```diff
@@ -1,15 +1,26 @@
...
+If you just want to use SLING from Python, e.g. want to try out the parser on a 
+pre-trained model, you can install the SLING wheel with pip. On a Linux machine
 with Python 3 you can install a pre-built wheel:
 
 ```
 sudo -H pip3 install http://www.jbox.dk/sling/sling-2.0.0-py3-none-linux_x86_64.whl
 ```
-and download the pre-trained model:
+
+You can test the installing by trying to import the `sling` package:
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply doc/guide/install.md]: Done

File: doc/guide/wikiflow.md [hunk 6]
```diff
@@ -289,25 +300,19 @@ Q2534120: {
...
 
 Once the Wikipedia documents have been parsed for all the languages you need,
 the information from these documents are collected into a Wikipedia item for
 each entity.
 
+All the links in the Wikipedia documents are collected and turned into a link
+graph that is stored in the `/w/item/links` property for each item. The link
+graph is built over all the Wikipedias being processed. The fan-in,
+i.e. the number of links to the item, is also computed and stored in the
+`/w/item/popularity` property. Tge popularity count also includes the number of
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply doc/guide/wikiflow.md]: Done.

File: doc/guide/wikiflow.md [hunk 7]
```diff
@@ -289,25 +300,19 @@ Q2534120: {
...
+times the item is a fact taget in other items.
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply doc/guide/wikiflow.md]: Done.

File: sling/nlp/wiki/wikipedia-links.cc (hunk 1 above)
Comment (ringgaard) [reply sling/nlp/wiki/wikipedia-links.cc]: Oops. Removed.

File: sling/nlp/wiki/wikipedia-links.cc (hunk 2 above)
Comment (ringgaard) [reply sling/nlp/wiki/wikipedia-links.cc]: Fixed.

**Pull request merged**